Add driver for bh1745 ambient light sensor

gblach opened the pull request.

gblach committed:
drivers: sensor: bh1745: add driver for bh1745 ambient light sensor

This driver is based on driver for bh1749 sensor from nRF Connect SDK.

Signed-off-by: Grzegorz Blach <grzegorz@blach.pl>

gblach committed:
boards: arm: thingy52_nrf52832: add support for bh1745 ambient light sensor

Signed-off-by: Grzegorz Blach <grzegorz@blach.pl>

MaureenHelm commented on drivers/sensor/bh1745/CMakeLists.txt:
```diff
@@ -0,0 +1,11 @@
+zephyr_library()
+zephyr_library_sources(bh1745.c)
+zephyr_library_sources_ifdef(CONFIG_BH1745_TRIGGER bh1745_trigger.c)
+
```
Please remove the extra blank line

gblach replied:
Done

MaureenHelm commented on drivers/sensor/bh1745/bh1745.c:
```diff
@@ -0,0 +1,312 @@
+
+	LOG_DBG("Fetching sample...\n");
+
+	if (i2c_reg_read_byte(data->i2c, DT_REG_ADDR(DT_DRV_INST(0)),
```
Please use the dt variant for all i2c transactions (`i2c_reg_read_byte_dt` and friends)

gblach replied:
Done

MaureenHelm commented on drivers/sensor/bh1745/bh1745.c:
```diff
@@ -0,0 +1,312 @@
+
+static struct bh1745_data bh1745_data;
+
+DEVICE_DEFINE(bh1745, DT_INST_LABEL(0),
+	      bh1745_init, NULL, &bh1745_data, NULL,
+	      POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY, &bh1745_driver_api);
```
Please update this driver to support multiple instances and use `DT_INST_FOREACH_STATUS_OKAY`

gblach replied:
Done

MaureenHelm commented on drivers/sensor/bh1745/bh1745_trigger.c:
```diff
@@ -0,0 +1,198 @@
+	struct bh1745_data *drv_data =
+		CONTAINER_OF(cb, struct bh1745_data, gpio_cb);
+
+	gpio_pin_interrupt_configure(dev, DT_INST_GPIO_PIN(0, int_gpios),
```
Please use the dt variant for all gpio transactions (`gpio_pin_interrupt_configure_dt` and friends)

gblach replied:
Done

MaureenHelm reviewed (changes requested):
Thank you for the contribution!

Please add this sensor to `tests/drivers/build_all/sensor/` to ensure it builds in CI.

gblach committed:
Merge branch 'zephyrproject-rtos:main' into main

gblach committed:
Merge branch 'zephyrproject-rtos:main' into main

gblach committed:
Merge branch 'zephyrproject-rtos:main' into main

gblach committed:
drivers: bh1745: re-work to follow current best practices

Update to follow best practices from upstream Zephyr, such as:

- use i2c_dt_spec and gpio_dt_spec to avoid boilerplate and save RAM
- log and propagate error codes received from drivers
- support multiple BH1749 device instances

Signed-off-by: Grzegorz Blach <grzegorz@blach.pl>

gblach committed:
Merge branch 'main' of github.com:gblach/zephyr

gblach committed:
tests: drivers: build_all: Build BH1745

Include BH1745 in the build all test.

Signed-off-by: Grzegorz Blach <grzegorz@blach.pl>

gblach committed:
Remove the extra blank line

Signed-off-by: Grzegorz Blach <grzegorz@blach.pl>

gblach committed:
Remove the extra blank line

gblach committed:
Merge branch 'main' of github.com:gblach/zephyr

gblach committed:
tests: drivers: build_all: sensor: Remove trailing whitespaces.

Signed-off-by: Grzegorz Blach <grzegorz@blach.pl>

gblach committed:
Merge branch 'main' into main
